alibi/prototypes/protoselect.py

```diff
@@ -66,60 +72,76 @@ def __init__(self,
         self.meta['params'].update({
             'kernel_distance': kernel_distance_tag,
             'eps': eps,
-            'lbd': lbd,
+            'lambda_penalty': lambda_penalty,
             'batch_size': batch_size,
             'verbose': verbose
         })
 
-    def fit(self,
-            X: np.ndarray,
-            X_labels: Optional[np.ndarray] = None,
-            Y: Optional[np.ndarray] = None) -> 'ProtoSelect':
+    def fit(self,  # type: ignore[override]
+            X_ref: Union[list, np.ndarray],
+            Y_ref: Optional[np.ndarray] = None,
+            X: Optional[Union[list, np.ndarray]] = None) -> 'ProtoSelect':
         """
-        Fit the explainer by setting the reference dataset.
+        Fit the summariser by setting the reference dataset. This step forms the kernel matrix in memory
+        which has a shape of `Nx x Ny`, where `Nx` is the number of instances in `X` and `Ny` is the
+        number of instances in `Y`.
 
         Parameters
         ---------
+        X_ref
+            Reference dataset to be summarised.
+        Y_ref
+            Labels of the reference dataset. The labels are expected to be represented as integers `[0, 1, ..., L-1]`,
+            where `L` is the number of classes in the reference dataset.
         X
-            Reference dataset to be summarized.
-        X_labels
-            Labels of the reference dataset.
-        Y
             Dataset to choose the prototypes from. If ``None``, the prototypes will be selected from the reference
-            dataset `X`.
+            dataset `X_ref`.
```

Comment:
I still find it slightly confusing because of the different behaviour whether `X` is present or not.

For instance, if `X` is present then the docstring for `X_ref` seems inaccurate (it would be `X` that is summarized?).

Also wording could be confusing, I assume `dataset to be summarised` and `dataset to choose the prototypes from` mean the same thing (i.e. "summarize" == "choose prototypes"?).

Finally, it would probably be helpful to include some of this from the method page in the docstrings too: 
![image](https://user-images.githubusercontent.com/13080878/167110412-f13fbb6f-c702-4505-8b36-d96160673f1d.png)